File: returned/python/today21/implement_queue_using_stacks_test.py
from implement_queue_using_stacks import MyQueue

def test1():
    myQueue = MyQueue()
    myQueue.push(1)
    myQueue.push(2)
    assert myQueue.peek() == 1
    assert myQueue.pop() == 1
    assert myQueue.empty() == False

def test2():
    myQueue = MyQueue()
    myQueue.push(1)
    myQueue.push(2)
    assert myQueue.peek() == 1
    myQueue.push(3)
    assert myQueue.peek() == 1

def test3():
    myQueue = MyQueue()
    myQueue.push(1)
    myQueue.push(8)
    assert myQueue.pop() == 1
    myQueue.push(3)
    assert myQueue.peek() == 8
    assert myQueue.pop() == 8
    myQueue.push(5)
    assert myQueue.pop() == 3

File: csharp/LeetTests/Today1/ImplementQueueUsingStacksTest.cs
using Today1.ImplementQueueUsingStacks;
namespace LeetTests.Today1;

internal class ImplementQueueUsingStacksTest
{
    [Test]
    public void TestImplementQueueUsingStacks()
    {
        MyQueue myQueue = new MyQueue();
        myQueue.Push(1); // queue is: [1]
        myQueue.Push(2); // queue is: [1, 2] (leftmost is front of the queue)
        Assert.That(myQueue.Peek(), Is.EqualTo(1)); // return 1
        Assert.That(myQueue.Pop(), Is.EqualTo(1)); // return 1, queue is [2]
        Assert.That(myQueue.Empty(), Is.EqualTo(false)); // return false
    }

    [Test]
    public void TestImplementQueueUsingStacks2()
    {
        //["MyQueue","push","push","peek","push","peek"]
        //[    [],     [1],   [2],   [],    [3],   []]
        MyQueue myQueue = new MyQueue();
        myQueue.Push(1);
        myQueue.Push(2);
        Assert.That(myQueue.Peek(), Is.EqualTo(1));
        myQueue.Push(3);
        Assert.That(myQueue.Peek(), Is.EqualTo(1));
    }
}